hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/test/AssertionSinks.java

```diff
@@ -153,6 +153,20 @@ private AssertionSinks() {
      * used to terminate the job so that you can {@code join()} it. This also
      * requires that there are no other assertions in the job as this one can
      * complete the job before the other ones succeeded.
+     * <p>
+     * The assertion can be validated as follows:
+     * <pre>{@code
+     * try {
+     *     jetInstance.newJob(p).join();
+     *     Assert.fail("Job should have completed with an AssertionCompletedException, " +
+     *             "but completed normally");
+     * } catch (CompletionException e) {
+     *     String errorMsg = e.getCause().getMessage();
+     *     Assert.assertTrue(
+     *             "Job failed with another reason then AssertionCompletedException",
```

Comment:
"then" -> "than"
We should also include the actual reason. Maybe `"Job was expected to complete with AssertionCompletedException, but completed with: " + e.getCause()`.